ONSdigital/rm-case-service
# 589 performance improvements to case service

# Motivation and Context
In the Case Service, it makes repeated calls to the collection exercise service of the same collex. Instead of wasting time repeating calls, we can make the method cacheable and instead it can use the cached version of the collection exercise if it's already there.

The other part of this ticket was to up the batch size of the case distributor and increase the messaging consumerThreads. Until we can use a performance environment, it's best to leave this bit and just add the cacheable to the Case service.

# What has changed
- Added `Cacheable` to `getCollectionExercise`
- Moved `application-test.yml` into the testing folder.


# How to test?
- Execute a large collection exercise and check how many times the `Retrieving collection exercise` log line gets printed.

# Links
[Trello](https://trello.com/c/2xfURaiX)
